src/osdc/Objecter.h

```diff
@@ -1793,7 +1793,7 @@ class Objecter : public md_config_obs_t, public Dispatcher {
 
     epoch_t last_force_resend = 0;
 
-    op_target_t(object_t oid, object_locator_t oloc, int flags)
+    op_target_t(const object_t& oid, const object_locator_t& oloc, int flags)
```

Comment:
Need to do the same for `hobject_t` constructors (`oid` and `nspace` arguments).  `op_target_t::get_hobj()` is called on every op -- does it show up in `_prepare_osd_op()` profile?

Reply:
In `_prepare_osd_op()` the final target for the created `hobject_t` is `MOSDOp.hobj`.
Currently, the `hobject_t` is first constructed in `op_target_t::get_hobj()`, then it is copied by the `MOSDOp` constructor.
So there is one copy of `hobject_t` which can be avoided here.
But I think that in order to avoid it we need to add an MOSDOp constructor taking `hobject_t&&`.

BTW, once we remove the std::string members from hobject_t and similar structs, those copies should become negligible.